src/lib/snark_work_lib/partitioned.ml

```diff
@@ -0,0 +1,331 @@
+open Core_kernel
+
+(* A `Pairing.t` identifies a single work in Work_selector's perspective *)
+module Pairing = struct
+  module UUID = struct
+    [%%versioned
+    module Stable = struct
+      module V1 = struct
+        (* this identifies a One_or_two work from Work_selector's perspective *)
+        type t = Pairing_UUID of int
+        [@@deriving compare, hash, sexp, yojson, equal]
+
+        let to_latest = Fn.id
+      end
+    end]
+  end
+
+  [%%versioned
+  module Stable = struct
+    module V1 = struct
+      (* Case `One` indicate no need to pair. This is needed because zkapp command
+         might be left in pool of half completion. *)
+      type t =
+        { one_or_two : [ `First | `Second | `One ]
+        ; pair_uuid : UUID.Stable.V1.t option
```

Comment:
What are semantics for this `option`? What does `{ one_or_two=`First; pair_uuid = None }` stand for?

Reply:
This make sense, I think only a `One could accept ID as None.

Reply:
This is the final version:

![image](https://github.com/user-attachments/assets/6e352afa-f92d-4661-ab74-500b107b983f)


In the WIP PR https://github.com/MinaProtocol/mina/pull/17076

Reply:
Do you want me to backport them?